Bottom: 6de5ea54cf3d9d46dde303144896f2a20cb06fea
Top:    b793c9e545b56ace6e749fe3919f1f6903958226
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-09 20:48:11 +0000

Refresh of make-the-strings-passed-to-att

---

diff --git a/src/misc.c b/src/misc.c
index b1d4304..34b18cb 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1892,7 +1892,7 @@ AttributeGetFromList (AttributeListType *list, char *name)
 }
 
 int
-AttributePutToList (AttributeListType *list, char *name, char *value, int replace)
+AttributePutToList (AttributeListType *list, const char *name, const char *value, int replace)
 {
   int i;
 
diff --git a/src/misc.h b/src/misc.h
index 82bd194..776e309 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -86,7 +86,7 @@ char *AttributeGetFromList (AttributeListType *list, char *name);
 /* Adds an attribute to the list.  If the attribute already exists,
    whether it's replaced or a second copy added depends on
    REPLACE.  Returns non-zero if an existing attribute was replaced.  */
-int AttributePutToList (AttributeListType *list, char *name, char *value, int replace);
+int AttributePutToList (AttributeListType *list, const char *name, const char *value, int replace);
 /* Simplistic version: Takes a pointer to an object, looks up attributes in it.  */
 #define AttributeGet(OBJ,name) AttributeGetFromList (&(OBJ->Attributes), name)
 /* Simplistic version: Takes a pointer to an object, sets attributes in it.  */
diff --git a/src/mymem.c b/src/mymem.c
index 59f8f8d..13863c1 100644
--- a/src/mymem.c
+++ b/src/mymem.c
@@ -718,7 +718,7 @@ MyRealloc (void *Ptr, size_t Size, const char *Text)
  * allocates memory for a new string, does some error processing
  */
 char *
-MyStrdup (char *S, const char *Text)
+MyStrdup (const char *S, const char *Text)
 {
   char *p = NULL;
 
diff --git a/src/mymem.h b/src/mymem.h
index 81dff7a..fdfd7e6 100644
--- a/src/mymem.h
+++ b/src/mymem.h
@@ -98,7 +98,7 @@ void **GetPointerMemory (PointerListTypePtr);
 void *MyCalloc (size_t, size_t, const char *);
 void *MyMalloc (size_t, const char *);
 void *MyRealloc (void *, size_t, const char *);
-char *MyStrdup (char *s, const char *);
+char *MyStrdup (const char *s, const char *);
 /* void MyFree (void **); */
 #define MYFREE(x) do { SaveFree(x); (x)=NULL; } while (0)
 void FreePolygonMemory (PolygonTypePtr);
